docs/configuring.rst

```diff
@@ -174,6 +175,14 @@ Miscellaneous options
     :type: string
     :default: ``#2980B9``
 
+.. confval:: flyout
```

Comment:
A config option `flyout` feels ambiguous to me -- that is, does `flyout=false` turn off the flyout entirely?

Maybe we want this to be `flyout_position` or `flyout_display` to mimic the CSS property?

- `flyout_display: floating`
- `flyout_display: hidden`
- `flyout_display: docked`

Does that make sense?

Reply:
> does `flyout=false` turn off the flyout entirely?

yes

> `flyout_display` to mimic the CSS property?

It makes sense, yes. For now, we will have only two options `hidden` (no flyout) and `navbar_bottom` (the current integration).

I suppose we will use this same config in the future to change the integration type: `navbar_top` or similar ones. Using a string will give us plenty of options here.

Reply:
So I assumed this was `flyout=false` would give the addons floating right flyout and `flyout=true` is the addons docked flyout in the navigation bar.

I'd probably avoid mixing positioning with `flyout_display`. I think we probably want a separate option for where the flyout is positioned: `flyout_display = "attached"; flyout_position = "bottom"` and `flyout_display = "floating"; flyout_position = "top right"`.

And we're only implementing `flyout_display = (hidden|attached)` for now. Or floating (default addons behavior) too?

I don't know what our plan is to control this from addons configuration though, it feels like there is some overlap, especially with floating positioning.

Reply:
You are right here, just with a small change, however. Note that with the theme configs we are controlling _only_ the flyout injected by the theme itself --we won't be controlling the Read the Docs flyout addons. That's controlled from the project's dashboard.

So, in the theme, we can implement as many display/position as we want, but those has to be implemented in the theme and won't be related with the different display/position we could eventually implement from Read the Docs addons.

Theme options won't have any effect on Read the Docs options. They are separated.

Reply:
Yeah, though this is getting back to what we have talked about with some intermediate customization pattern. I'm seeing this becoming more important for us to define, especially given we're not seeing many maintainers jump into full addons reimplementation yet.

We should continue the discussion about intermediate patterns:

- https://github.com/readthedocs/addons/issues/51

Floating positioning is an example here, but this probably is not super useful for the theme to redefine configuration for.

Reply:
Yeah, we should continue talking about that customization pattern. In this PR, I'm only wanting to have a setting to disable the "feature is enabled on Read the Docs automatically" and currently can't be disabled. So, building locally and building on Read the Docs leads to different results.

People wanting to use _only the flyout addons_ can't currently do that because the theme always injects the flyout in the navbar.

Reply:
To clarify, I feel like this is still the plan:

> And we're only implementing flyout_display = (hidden|attached) for now.

We can discuss intermediate customization patterns in https://github.com/readthedocs/addons/issues/51 and decide whether that means there is a `floating` option here as well.

Or rather, if it feels like `floating` might be an option here at some point, the above plan is more future forward than `flyout: (true|false)`.

Reply:
I updated the PR to use `flyout_display: (hidden|attached)` 👍🏼